app/controllers/users_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  before_action :find_users, except: %i(index new create)
+
+  def index; end
+
+  def show; end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      log_in @user
+      redirect_to @user
+      flash[:success] = 'User successfully created'
+    else
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @user.update(user_params)
+      redirect_to @user
+      flash[:success] = 'User successfully updated'
+    end
+  end
+
+  def destroy
```

Comment:
This action is not protected against unauthorized users so anybody can delete users just by figuring out proper `id` , same for edit and update.